morphia/src/main/java/org/mongodb/morphia/mapping/Mapper.java

```diff
@@ -737,7 +755,7 @@ private void addConverters(final MappedClass mc) {
     private MappedClass addMappedClass(final MappedClass mc, final boolean validate) {
         addConverters(mc);
 
-        if (validate) {
+        if (validate && !mc.isInterface()) {
```

Comment:
if you comment out the interface check and the `@Embededed` on `Nested` in `EmbeddedMappingTest` then run `org.mongodb.morphia.TestMapper#subTypes` you'll see the failure.